opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -254,7 +254,80 @@ namespace Opm
         return changed;
     }
 
+    template<typename TypeTag>
+    bool
+    WellInterface<TypeTag>::
+    updateWellControlAndStatusLocalIteration(const Simulator& ebos_simulator,
+                      WellState& well_state,
+                      const GroupState& group_state,
+                      const Well::InjectionControls& inj_controls,
+                      const Well::ProductionControls& prod_controls,
+                      const double& wqTotal, 
+                      DeferredLogger& deferred_logger) /* const */
+    {
+        const auto& summary_state = ebos_simulator.vanguard().summaryState();
+        const auto& schedule = ebos_simulator.vanguard().schedule();
+        
+        if (this->wellUnderZeroRateTarget(summary_state, well_state) || !(this->well_ecl_.getStatus() == WellStatus::OPEN)) {
+           return false;
+        }
 
+        const double sgn = this->isInjector() ? 1.0 : -1.0;
+        if (!this->wellIsStopped()){
+            if (wqTotal*sgn <= 0.0){
+                this->stopWell();
+                return true;
+            } else {
+                bool changed = false;
+                auto& ws = well_state.well(this->index_of_well_);
+                const bool hasGroupControl = this->isInjector() ? inj_controls.hasControl(Well::InjectorCMode::GRUP) :
+                                                                  prod_controls.hasControl(Well::ProducerCMode::GRUP);
+
+                changed = this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
+                if (hasGroupControl) {
```

Comment:
The logic here can be problematic. For example, if  `checkIndividualConstraints` is `true` and `checkGroupConstraints` is false. 

after the two calls, 
`changed` will false. 

Do you think we should change the line below to be 
`changed = changed  | this->checkGroupConstraints(well_state, group_state, schedule, summary_state, deferred_logger); 
`?
or we should rather use different if condition

`if (!changed  && hasGroupControl) ` to not call `checkGroupConstraints` ?

Reply:
You're absolutely right. My intention was to check both, so I'd prefer 
`changed = changed  | this->checkGroupConstraints(...)`

Reply:
> ```C++
> changed = changed || this->checkGroupConstraints(...)
> ```

Just remember that `||` has a short-circuiting behaviour so if `changed == true` then
```C++
changed || this->checkGroupConstraints(...)
```
will be unconditionally `true` and `this->checkGroupConstraints()` will not be called.  Is that okay here?

Reply:
Maybe we should do the following:

```c++
                const bool individual_changed = this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
               const bool group_changed = hasGroupControl && this->checkGroupConstraints(well_state, group_state, schedule, summary_state, deferred_logger);

              changed = individual_changed || group_changed;
```

Reply:
The complication is that both `checkIndividualConstraints` and `checkGroupConstraints` have actions involved and not purely checking.